src/main.rs

```diff
@@ -89,7 +95,13 @@ fn build_subcommands<'a, 'b>(
     subcommands: &'a Vec<Command>,
 ) -> App<'a, 'b> {
     for c in subcommands {
-        let mut subcmd = SubCommand::with_name(&c.name).about(c.desc.as_ref());
+        let mut subcmd = SubCommand::with_name(&c.name)
+            .about(c.desc.as_ref())
+            .setting(if atty::is(Stream::Stdout) {
+                AppSettings::ColoredHelp
+            } else {
+                AppSettings::ColorNever // in case it run on terminal without color support
+            });
```

Comment:
In my testing a few weeks ago, i thought the `ColoredHelp` setting was inherited from the root `cli_app`? We actually need to do this here too?

Reply:
Unfortunately yes. Seems like most of `AppSettings::*` in clap doesn't propagate to Subcommands (or in another word, it doesn't applied to nested `App`). Probably it propagate to Subcommands only if the setting prefixed with `*Subcommands` (e.g `VersionlessSubcommands`) 🤔

<details><summary>example if <code>ColoredHelp</code> only applied to root <code>cli_app</code></summary>

![Screenshot_20190918_021758](https://user-images.githubusercontent.com/4953069/65072083-613c5880-d9ba-11e9-819b-6c64e3c4ead5.png)
</details>